cpp/oneapi/dal/algo/knn/common.hpp

```diff
@@ -72,6 +76,35 @@ using v1::by_default;
 
 } // namespace method
 
+/// Represents result option flag
+/// Behaves like a regular :expr`enum`.
+class result_option_id : public result_option_id_base {
+public:
+    result_option_id() : result_option_id_base{} {}
+    result_option_id(const result_option_id_base& base) : result_option_id_base{ base } {}
```

Comment:
Why did you introduce this constructor?

Reply:
There is a dependency in `common.hpp` when we convert from base to the derived type

Reply:
Why conversion from base to the derived type is needed?